src/settings/index.ts

```diff
@@ -0,0 +1,95 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as semver from 'semver';
+import { vscodePath as getAppPath } from './vscodePath';
+import { version as extVersion, iconSuffix, customIconFolderName } from './extSettings';
+import { ISettings } from '../models/ISettings';
+import { IState } from '../models/IState';
+import { IVSCode } from '../models/IVSCode';
+
+export interface IExtensionStatus {
+  enabled: string;
+  disabled: string;
+  notInstalled: string;
+}
+
+export interface ISettingsManager {
+  getSettings: () => ISettings;
+  getState: () => IState;
+  setState: (state: IState) => void;
+  setStatus: (sts: string) => void;
+  deleteState: () => void;
+}
+
+export const status: IExtensionStatus = {
+  enabled: 'enabled',
+  disabled: 'disabled',
+  notInstalled: 'notInstalled',
+};
+
+export class SettingsManager {
+  private settings: ISettings;
+
+  constructor(private vscode: IVSCode) { }
+
+  public getSettings(): ISettings {
+    if (this.settings) { return this.settings; };
+    const isInsiders = /insiders/i.test(this.vscode.env.appName);
+    const version = semver(this.vscode.version);
+    const isGt160 = semver.lt(version.major + '.' + version.minor + '.' + version.patch, '1.6.0');
```

Comment:
I think there is a bug here. In order for `isGt160` to be `true` `semver.gt` function needs to be used. Am I missing something?